homeassistant/components/recorder/statistics.py

```diff
@@ -2029,42 +2032,67 @@ def _generate_statistics_at_time_stmt(
     metadata_ids: set[int],
     start_time_ts: float,
     types: set[Literal["last_reset", "max", "mean", "min", "state", "sum"]],
+    slow_dependant_subquery: bool,
 ) -> StatementLambdaElement:
     """Create the statement for finding the statistics for a given time."""
-    # This query is the result of significant research in
-    # https://github.com/home-assistant/core/issues/132865
-    # A reverse index scan with a limit 1 is the fastest way to get the
-    # last start_time_ts before a specific point in time for all supported
-    # databases. Since all databases support this query as a join
-    # condition we can use it as a subquery to get the last start_time_ts
-    # before a specific point in time for all entities.
+
     stmt = _generate_select_columns_for_types_stmt(table, types)
-    stmt += (
-        lambda q: q.select_from(StatisticsMeta)
-        .join(
-            table,
+    if slow_dependant_subquery:
+        # Simple group-by for MySQL, must use less
+        # than 1000 metadata_ids in the IN clause for MySQL
+        # or it will optimize poorly.
```

Comment:
Same question, should we guard against too many metadata_ids?

Reply:
f3828a622eb